### PR TITLE
task6 - work with DB

### PR DESCRIPTION
**Pull request by Aksana-Tsishchanka**

### CODE REVIEW

File: seeders/20171121050656-insert-users.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.bulkInsert('Users', [{
+      email: 'test4@gmail.com',
+      pass: '123'
```
Comment (boccob) [inline seeders/20171121050656-insert-users.js:7]: Not the best way to store passwords.

File: database/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      .authenticate()
+      .then(() => {
+        console.log('Connection has been established successfully.');
+      })
+      .catch(err => {
+        console.error('Unable to connect to the database:', err);
+      });
+  }
+  
+  static createUserTable() {
```
Comment (boccob) [inline database/index.js]: A bit strange to have models definitions in two places.

File: database/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import Sequelize from 'sequelize';
+
+export default class DataBase {
+  static sequelize;
+  static init() {
```
Comment (boccob) [inline database/index.js]: It's async function I'd say

File: database/index.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      email: {
+        type: Sequelize.STRING,
+        unique: true,
+      },
+      pass: {
+        type: Sequelize.STRING
+      }
+    });
+    
+    this.userTable.sync({ force: true }).then(() => {
```
Comment (boccob) [inline database/index.js]: What if I don't want to drop the table at every start of application?

File: app.js [hunk 5]
```diff
@@ -48,49 +50,52 @@ app.post('/auth', jwtAuth);
...
+  try {
+    user = await DB.getUsers();
+  }
+  catch(e) {
+    response.json({ error: e });
+  }
+  response.json(user);
 });
 
 /*passport local strategy */
```
Comment (boccob) [inline app.js:108]: Why don't you use data from database for authorization?

File: app.js [hunk 6]
```diff
@@ -48,49 +50,52 @@ app.post('/auth', jwtAuth);
...
+      res.send({ error: "product can't be created" });
+    }
     res.send(newProduct);
   });
 
-function getProduct(id) {
-  const product = getModelById(products, id);
-  if (product) {
-    return product;
+router.get('/api/products/:id', async (request, response) => {
```
Comment (boccob) [inline app.js:84]: It doesn't work. Probably because you don't cast string to number.

File: database/index.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  }
+  
+  /*
+  product [{
+    id,
+    name,
+  }]
+   */
+  static createProductTable(products) {
+    this.productTable = this.sequelize.define('Products', {
```
Comment (boccob) [inline database/index.js:52]: Did you create reviews before? I suggest to create a separate model and check how relations work in sequelize.

File: app.js (hunk 6 above)
Comment (Aksana-Tsishchanka) [reply app.js:84]: Thx! fixed. The problem was wrong naming in static async getProductById.

File: database/index.js (hunk 3 above)
Comment (Aksana-Tsishchanka) [reply database/index.js]: thx, fixed

File: database/index.js (hunk 2 above)
Comment (Aksana-Tsishchanka) [reply database/index.js]: I could not find the second place. Another user model is used for migration.

File: database/index.js (hunk 4 above)
Comment (Aksana-Tsishchanka) [reply database/index.js]: in that case this.userTable.sync() will resolve it.

File: app.js (hunk 5 above)
Comment (Aksana-Tsishchanka) [reply app.js:108]: good point

File: seeders/20171121050656-insert-users.js (hunk 1 above)
Comment (Aksana-Tsishchanka) [reply seeders/20171121050656-insert-users.js:7]: agreed, in future, I will store a hashed password.

File: database/index.js (hunk 2 above)
Comment (boccob) [reply database/index.js]: database/index.js and models/user.js

**Review by boccob — APPROVED** [review]

**Pull request merged**